oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/authentication/logout/OidcBackchannelLogoutFilter.java

```diff
@@ -0,0 +1,206 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.web.authentication.logout;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.security.oauth2.client.oidc.authentication.logout.OidcLogoutToken;
+import org.springframework.security.oauth2.client.oidc.authentication.session.InMemoryOidcProviderSessionRegistry;
+import org.springframework.security.oauth2.client.oidc.authentication.session.OidcProviderSessionRegistrationDetails;
+import org.springframework.security.oauth2.client.oidc.authentication.session.OidcProviderSessionRegistry;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.jwt.BadJwtException;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtDecoderFactory;
+import org.springframework.security.web.authentication.logout.BackchannelLogoutAuthentication;
+import org.springframework.security.web.authentication.logout.BackchannelLogoutHandler;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * A filter for the Client-side OIDC Backchannel Logout endpoint
+ *
+ * @author Josh Cummings
+ * @since 6.1
+ * @see <a target="_blank" href=
+ * "https://openid.net/specs/openid-connect-backchannel-1_0.html">OIDC Backchannel Logout
+ * Spec</a>
+ */
+public class OidcBackchannelLogoutFilter extends OncePerRequestFilter {
+
+	private final Log logger = LogFactory.getLog(getClass());
+
+	private static final String ERROR_MESSAGE = "{ \"error\" : \"%s\", \"error_description\" : \"%s\" }";
+
+	private final ClientRegistrationRepository clientRegistrationRepository;
+
+	private final JwtDecoderFactory<ClientRegistration> logoutTokenDecoderFactory;
+
+	private RequestMatcher requestMatcher = new AntPathRequestMatcher("/oauth2/{registrationId}/logout", "POST");
```

Comment:
See other comment regarding the patter naming -> `/connect/back-channel-logout/{registrationId}`

Reply:
What is the rationale behind `/connect` vs `/oidc`?

Reply:
`/connect` is convention used. See [OpenID Provider Configuration Response](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse)

Reply:
@jzheaux You mentioned the reason for using `/logout/connect/back-channel/{registrationId}` instead of `/connect/logout/{registrationId}`

> The reason for that is it appears to follow the naming convention for other logout URLs

Can you provide more details?

Reply:
Sure.

The first is Spring Security's local logout, which is `/logout`. Another is SAML 2.0's logout endpoint, which is `/logout/saml2/slo`. Incidentally, this is also the case with `/login` for local, SAML 2.0, and OAuth 2.0 login endpoints.

Since I didn't make most of those decisions, I checked with Rob for historical context, and here is what he offered:

> It's the folder vs tags problem really. It's both saml and login. I chose login and logout because it helps when users aren't even sure what type of authentication is happening
>
> One could argue the other way around, but I am rather convinced that there is not enough value in changing the pattern at this point
>
> I would stick with `/logout/*` and we can always allow customization

I think the above statement about conventionality is enough for me to go with `/logout/connect/back-channel`. However, in case it is also helpful, one place where we `/logout/connect/back-channel` is nice is metrics. I personally see greater value in drilling down this way: I want stats on all logout behavior (`/logout/*`), then maybe I drill down to stats on logout strategies after that. It's nice that `/logout/connect/back-channel` reflects that analytics mentality.